File: app/src/main/java/me/smbduknow/transport/geo/LocationProvider.java
package me.smbduknow.transport.geo;

import android.location.Location;

public interface LocationProvider {

    interface OnRecieveLocationListener {
        void onProviderConnected();
        void onReceiveLocation(Location location);
        void onReceiveFailed();
    }

    boolean isAvailable();

    void requestLastLocation();
    void requestLocationUpdates();
    void removeLocationUpdates();

//    void release();
}

File: app/src/main/java/me/smbduknow/transport/geo/LocationProvider.kt
package me.smbduknow.transport.geo

import android.location.Location

interface LocationProvider {

    interface OnRecieveLocationListener {
        fun onProviderConnected()
        fun onReceiveLocation(location: Location)
        fun onReceiveFailed()
    }

    val isAvailable: Boolean

    fun requestLastLocation()
    fun requestLocationUpdates()
    fun removeLocationUpdates()

    //    void release();
}
